src/main/java/org/assertj/core/util/Preconditions.java

```diff
@@ -61,24 +62,11 @@ public static CharSequence checkNotNullOrEmpty(CharSequence s, String message) {
    * @throws IllegalArgumentException if the given array is empty.
    */
   public static <T> T[] checkNotNullOrEmpty(T[] array) {
-    T[] checked = checkNotNull(array);
+    T[] checked = requireNonNull(array);
     if (checked.length == 0) throwExceptionForBeingEmpty();
     return checked;
   }
 
-  /**
-   * Verifies that the given object reference is not {@code null}.
-   * 
-   * @param <T> the type of the reference to check.
-   * @param reference the given object reference.
-   * @return the non-{@code null} reference that was validated.
-   * @throws NullPointerException if the given object reference is {@code null}.
-   */
-  public static <T> T checkNotNull(T reference) {
```

Comment:
Maybe we have to keep the method around to avoid breaking binary compatiblity as the class is not part of the `internal` package?

Reply:
I would keep it but marking as deprecated and then it could be dropped with the next major release

Reply:
agreed, along with indicating the alternative in the deprecated javadoc tag description